server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -258,6 +255,10 @@ public void sendShuffleData(SendShuffleDataRequest req,
           break;
         }
       }
+      // since the required buffer id is only used once, the shuffle client would try to require another buffer whether
+      // current connection succeeded or not. Therefore, the preAllocatedBuffer is first get and removed, then after
+      // cacheShuffleData finishes, the preAllocatedSize should be updated accordingly.
+      manager.releasePreAllocatedSize(info.getRequireSize());
```

Comment:
`info` may be null?

Reply:
Do we release `allocateMemory` in any other places? Is there conflicts if we release the memory here?

Reply:
OK. @xianjingfeng Do you have another suggestion?